api/capi/include/nntrainer-tizen-internal.h

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   nntrainer-tizen-internal.h
+ * @date   30 June 2021
+ * @brief  NNTrainer CAPI header for tizen interanl api.
+ * @note   This header is designed to be used only in Tizen
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#ifndef __NNTRAINER_TIZEN_INTERNAL_H__
+#define __NNTRAINER_TIZEN_INTERNAL_H__
+
+#include <nntrainer.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+/**
+ * @brief Save the model after training
+ * @details Use this function to save the current model. @a format rules
+ * how and which part of the model should be saved. Some formats may save
+ * parameters required for training. Some other formats may save model
+ * configurations. The model must be compiled and has allocated memory before
+ * calling this function.
+ * @see @a ml_train_model_format_e to check which part of the model is
+ * saved
+ * @note This function overrides the existing file without any notification.
+ *
+ * @since_tizen 6.5
+ * @param[in] model The NNTrainer model handle to save
+ * @param[in] file_path File path to save the file.
```

Comment:
is this going to be file path or file prefix?

Reply:
It will be a single file. (if there are two file, one file will point to another one)

Reply:
so the file extension must be provided by the user then? how about we decide it based on the format?

Reply:
I think it's commonly expected that user puts file path including the extension. Toco, pytorch, flatbuffers does not include extensions.

Reply:
then our implementation will check if the extension matches the format (.bin for BIN and .ini for INI/INI+BIN)? if yes, add to documentation for the expected format.

Reply:
I think we don't need to, and should check or force the extension, as other's also support this, analogically, as we can edit something in notepad and can save as `.html`.

Adding this restriction is not to hard thing to do though :)

Reply:
Yes, IMO the extension forcing just makes things easier to read.
@jijoongmoon what do you think?

Reply:
After resolving this, i guess we can merge this?

Reply:
I guess we need to save with the name given by user. One thing we need to make sure is that this is for saving training result. not for the inference. We might have to have another api for inference later.. ( maybe after finishing tflite output generating )